### PR TITLE
docs: devstack settings

### PR DESCRIPTION
**Pull request by regisb**
**Description:**

Plugins may have a "devstack" setting, and this was not documented (as far as I know). I was bitten by this as I was trying to override settings in a plugin, and my settings were being overridden by devstack settings.

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @regisb!

### What's next?

*Please work through the following steps to get your changes ready for engineering review:*

#### :radio_button: Get product approval

If you haven't already, [check this list](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3875962884/How+to+submit+an+open+source+contribution+for+Product+Review#Does-my-contribution-require-Product-Review%3F) to see if your contribution needs to go through the product review process.

- If it does, you'll need to submit a product proposal for your contribution, and have it reviewed by the [Product Working Group](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3449028609/Product+Working+Group).
    - This process (including the steps you'll need to take) is documented [here](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3875962884/How+to+submit+an+open+source+contribution+for+Product+Review#Product-Review-Process).
- If it doesn't, simply proceed with the next step.

#### :radio_button: Provide context

To help your reviewers and other members of the community understand the purpose and larger context of your changes, feel free to add as much of the following information to the PR description as you can:

- Dependencies
  > This PR must be merged before / after / at the same time as ...
- Blockers
  > This PR is waiting for OEP-1234 to be accepted.
- Timeline information
  > This PR must be merged by XX date **because** ...
- Partner information
  > This is for a course on edx.org.
- Supporting documentation
- Relevant [Open edX discussion forum](https://discuss.openedx.org/) threads





#### :radio_button: Get a green build

If one or more checks are failing, continue working on your changes until this is no longer the case and your build turns green.



#### :radio_button: Let us know that your PR is ready for review:

### Who will review my changes?


This repository is currently maintained by `@openedx/2u-arch-bom`.  Tag them in a comment and let them know that your changes are ready for review.


### Where can I find more information?

If you'd like to get more details on all aspects of the review process for open source pull requests (OSPRs), check out the following resources:

- [Overview of Review Process for Community Contributions](https://docs.openedx.org/en/latest/developers/references/developer_guide/process/FAQ-about-pull-requests.html)
- [Pull Request Status Guide](https://docs.openedx.org/en/latest/developers/references/developer_guide/process/pull-request-statuses.html)
- [Making changes to your pull request](https://docs.openedx.org/en/latest/documentors/how-tos/make_changes_to_your_pull_request.html)

### When can I expect my changes to be merged?

Our goal is to get community contributions seen and reviewed as efficiently as possible.

However, the amount of time that it takes to review and merge a PR can vary significantly based on factors such as:

- The size and impact of the changes that it introduces
- The need for product review
- Maintenance status of the parent repository

:bulb: *As a result it may take up to several weeks or months to complete a review and merge your PR.*

**Comment by robrap** [comment]
Thanks @regisb. I’ll take this to our teams because:
1. I’m not aware of us needing to do this for our plugin, and maybe that is just personal lack of awareness and maybe there are limited circumstances where this is required?
2. 2U is helping with a DEPR around the devstack.py files. See this relevant comment: https://github.com/openedx/public-engineering/issues/247#issuecomment-2344711918. It would be nice to understand how this all fits in with the plan.

That said, I understand your intent is simply to get the docs in line with reality as a step.

**Comment by regisb** [comment]
> I’m not aware of us needing to do this for our plugin, and maybe that is just personal lack of awareness and maybe there are limited circumstances where this is required?

We are currently working on a plugin ([the new forum](https://github.com/edly-io/forum/)) that needs to override the `COMMENTS_SERVICE_URL` -- including in development. I did add the updated setting in common.py, but these values are immediately overridden by the values from edx-platform's devstack.py. A [quick search](https://github.com/search?q=org%3Aopenedx+settings.devstack+&type=code) revealed that the devstack settings are already used in the event routing backend ([here](https://github.com/openedx/event-routing-backends/blob/42c9e8ed9159d23df27de13321904fbaeedaf433/event_routing_backends/apps.py#L22)).

> 2U is helping with a DEPR around the devstack.py files. See this relevant comment: https://github.com/openedx/public-engineering/issues/247#issuecomment-2344711918. It would be nice to understand how this all fits in with the plan.

I wasn't aware of that. I'll comment there.

**Comment by mphilbrick211** [comment]
Hi @robrap - any update on this?

**Comment by robrap** [comment]
@regisb: What is the status of this PR? Do you know how it relates to any changes that might be happening in terms of "devstack" => "development", or do we need to bring in Kyle? Thank you.

**Comment by regisb** [comment]
I am not familiar with the recent updates, so I'm not in the best position to comment. AFAIK this documentation change is still relevant.

**Comment by robrap** [comment]
@kdmccormick: Is this ready to land as-is? Will it need to be updated in the future? Thank you.

**Comment by kdmccormick** [comment]
We had planned to replace devstack.py with a simpler, YAML-free development.py. But, we found that Tutor's use of devstack.py is tightly coupled to production.py and the YAML-based settings schema. Replacing devstack.py without also replacing production.py would actually increase the overall complexity of the system. So, we are going to leave it as-is for now, and then introduce simpler YAML-free production+development settings all at once further down the road.

In short-- we will replace devstack.py eventually, but not in the immediate future. It would be reasonable to add this "devstack" slot to Django App Plugins now, with the understanding that we'll eventually need to deprecate it and replace it with a "development" slot.

**Review by robrap — APPROVED** [review]
